packages/workflow/src/Interfaces.ts

```diff
@@ -1740,6 +1680,10 @@ export type AllEntities<M> = M extends { [key: string]: string } ? Entity<M, key
 
 export type Entity<M, K> = K extends keyof M ? { resource: K; operation: M[K] } : never;
 
+type RequiredProps<F, Required extends keyof F> = F & {
+	[Key in Required]-?: F[Key];
+};
```

Comment:
I'd maybe rename the type to clarify the purpose (since it looks similar to the built-in `Required`), and the second generic because it is not `Required` when passed in, but rather made required by the transformation.

```suggestion
type MakeOptionalKeysRequired<O, OptionalKey extends keyof O> = O & {
	[K in OptionalKey]-?: O[K];
};
```

Reply:
You could also restrict the second generic so only optional keys can be passed in to be made required.

Reply:
I created a less generic type called `FunctionsBaseWithRequiredKeys`.